subiquity/server/apt.py

```diff
@@ -264,7 +264,7 @@ async def _restore_dir(dir):
 
         await self.unmount(
                 Mountpoint(mountpoint=target_mnt.p('cdrom')),
-                remove=False)
+                remove=True)
```

Comment:
Can't we just remove this unmount call entirely now? And move the rmdir after the call to cleanup()?